surfactant/filetypeid/id_magic.py

```diff
@@ -28,7 +28,36 @@ def identify_file_type(filepath: str) -> Optional[str]:
             if magic_bytes[:4] == b"\x7fELF":
                 return "ELF"
             if magic_bytes[:2] == b"MZ":
+                # Check to see if this is actually a PE file or a DOS file
+                coff_addr = (
+                    int.from_bytes(magic_bytes[0x3C:0x40], byteorder="little", signed=False)
+                    & 0xFFFF
+                )
+
+                # Check to see if the coff_addr is still within the initial
+                # read; if not, we need at least another one, maybe more.
+                # Unlikely, but possible if the PE/file is malformed.
+                if coff_addr > len(magic_bytes):
+                    magic_bytes += f.read(265)
```

Comment:
```suggestion
                if coff_addr+4 > len(magic_bytes):
                    magic_bytes += f.read(coff_addr+4-len(magic_bytes))
```

Since this is likely to already be an odd/malformed PE file, we might as well attempt a read from the file that gets all of the magic bytes for the PE header.